plotting/basemap.py

```diff
@@ -1,120 +1,98 @@
-                    ellps="WGS84",
-                    projection=projection,
-                    boundinglat=min_lat,
-                    lon_0=center[1],
-                )
-            elif projection == "merc":
-                basemap = Basemap(
-                    resolution="c",
-                    # area_thresh=((height*width)/(1000*1000))*0.00001 , #display islands whose area is 0.001% of displayed area
-                    ellps="WGS84",
-                    projection=projection,
-                    llcrnrlat=height[0],
-                    llcrnrlon=width[0],
-                    urcrnrlat=height[1],
-                    urcrnrlon=width[1],
-                )
-            else:
-                basemap = Basemap(
-                    resolution=get_resolution(height, width),
-                    area_thresh=((height * width) / (1000 * 1000))
-                    * 0.00001,  # display islands whose area is 0.001% of displayed area
-                    ellps="WGS84",
-                    projection=projection,
-                    lat_0=center[0],
-                    lon_0=_convert_to_bounded_lon(center[1]),
-                    height=height,
-                    width=width,
-                )
-            basemap.filename = filename
-
-            def do_pickle(basemap: Basemap, filename: str) -> None:
-                pickle.dump(basemap, open(filename, "wb"), -1)
+            fig = plt.figure(figsize=figuresize, dpi=dpi)
+
+            map_plot = plt.axes(projection=plot_proj, facecolor="dimgrey")
+            map_plot.set_extent(extent, crs=plot_proj)
+
+            map_plot.add_geometries(
+                land_shp.geometries(),
+                crs=ccrs.PlateCarree(),
+                facecolor="grey",
+                edgecolor="black",
+            )
+
+            map_plot.gridlines(
+                draw_labels={"bottom": "x", "left": "y"},
+                dms=True,
+                x_inline=False,
+                y_inline=False,
+                xlabel_style={"size": 10, "rotation": 0},
+                ylabel_style={"size": 10},
+                zorder=2,
+            )
+
+            def do_pickle(fig, ax, filename: str) -> None:
+                pickle.dump((fig, ax), open(filename, "wb"), -1)
 
             if not os.path.isdir(CACHE_DIR):
                 os.makedirs(CACHE_DIR)
 
-            t = threading.Thread(target=do_pickle, args=(basemap, filename))
+            t = threading.Thread(target=do_pickle, args=(fig, map_plot, filename))
             t.daemon = True
             t.start()
-
-        _maps_cache[filename] = basemap
     else:
-        basemap = _maps_cache[filename]
-
-    return basemap
-
-
-def load_nwatlantic() -> return_type(load_map):
-    return load_map("lcc", (55, -60), 5e6, 3.5e6)
-
-
-def load_arctic() -> return_type(load_map):
-    return load_map("npstere", (65, 0), None, None)
-
-
-def load_pacific() -> return_type(load_map):
-    return load_map("lcc", (53, -137), 2e6, 2.5e6)
```

Comment:
I couldn't find a place in the code where these functions were used so they were dropped. If they're necessary just let me know and I'll restore them.

Reply:
keep em gone